### PR TITLE
JSp validations

### PR DESCRIPTION
**Pull request by amoldp**

### CODE REVIEW

File: src/main/java/com/asu/cse545/group12/domain/Transactions.java [hunk 1]
```diff
@@ -16,7 +17,7 @@
 	private int transactionId;
 	
 	
-
+	@Pattern(regexp="^[a-zA-Z0-9äöüÄÖÜ]*$", message="Enter correct username")
```
Comment (chasethenag420) [inline src/main/java/com/asu/cse545/group12/domain/Transactions.java]: Should be of type integer

File: src/main/webapp/WEB-INF/jsp/login.jsp [hunk 2]
```diff
@@ -107,15 +123,17 @@
...
           <input id="keyboard" class="form-control" type='text' name='username' value=''>
+
           </span> 
           <br> <br> 
           <label class="col-sm-4">
             <spring:message	code="label.form.loginPass"> </spring:message>
           </label>
           <span class="col-sm-8"> <input class="form-control"
-            id="keyboard" type='password' name='password' />
+ss            id="keyboard" type='password' name='password' />
```
Comment (chasethenag420) [inline src/main/webapp/WEB-INF/jsp/login.jsp:133]: Remove this.

Comment (ganeshzilpe) [inline src/main/webapp/WEB-INF/jsp/login.jsp:133]: This is added by Saranya for virtual keyboard. Why do u want to remove it??

Comment (chasethenag420) [inline src/main/webapp/WEB-INF/jsp/login.jsp:133]: I asked to remove ss at the start of line

**Comment by chasethenag420** [comment]
@amoldp  I am somehow not convinced with this approach of validating the bean class because the same bean object we are using in different forms and some fields may come as null from some screens. I would say best would be to add something like TransactionInputValidator.java which will be used in TransactionContoller.java. Take a look at them any do it similar way.  @ganeshzilpe any thoughts on this?

**Comment by amoldp** [comment]
@chasethenag420, can you please give me an example?

**Comment by amoldp** [comment]
@chasethenag420 I can already see external classes for create user and
transactions.

**Comment by ganeshzilpe** [comment]
If we are having validator classes for those domain classes, then I don't
think there will be any problem. But if we do annotations validation in
classes like form, search form or transactionform classes where other
fields may come empty, it will give problem.

Note: validation for username and password should not be done for null or
empty check in users class. Because we are using same for create employee
where system administrator won't put username and password for creating
internal user.
So if someone put validation there is user class, plz remove it and
validate in createexternaluservalidation.java

Thabks

**Pull request merged**